scalalib/src/mill/javalib/android/AndroidAppBundle.scala

```diff
@@ -0,0 +1,214 @@
+  }
+
+  /**
+   * Packages compiled resources and dex files into a zip for app bundle generation.
+   *
+   * @return PathRef to the created bundle zip.
+   */
+  def androidBundleZip: T[PathRef] = Task {
+    val dexFile = androidDex().path
+    val baseDir = Task.dest / "base"
+
+    // Create directory structure for Android bundle
+    Seq("manifest", "dex").foreach(d => os.makeDir.all(baseDir / d))
+
+    // Copy dex and resource files into the bundle directory
+    os.copy(dexFile, Task.dest / dexFile.last)
+    os.unzip(Task.dest / os.up / "androidResources.dest" / "res.apk", Task.dest)
+    os.copy(Task.dest / "AndroidManifest.xml", baseDir / "manifest" / "AndroidManifest.xml")
+
+    // Copy .pb files (compiled protobuf resources) and organize dex files
+    os.walk(Task.dest).filter(_.ext == "pb").foreach(file => os.copy(file, baseDir / file.last))
+    os.walk(Task.dest).filter(_.ext == "dex").zipWithIndex.foreach { case (file, idx) =>
+      os.copy(file, baseDir / "dex" / (if (idx == 0) "classes.dex" else s"classes${idx + 1}.dex"))
+    }
+
+    // Copy additional resources (res, lib, assets) if present
+    Seq("res", "lib", "assets").foreach { folder =>
+      val folderPath = Task.dest / folder
+      if (os.exists(folderPath)) {
+        os.copy.over(folderPath, baseDir / folder)
+      }
+    }
+
+    // Create the final bundle zip
+    os.zip(Task.dest / "bundle.zip", Seq(baseDir))
+
+    PathRef(Task.dest / "bundle.zip")
+  }
+
+  /**
+   * Creates an Android App Bundle (AAB) using bundle tool from the bundle zip.
+   *
+   * @return PathRef to the created app bundle (AAB).
+   */
+  def androidBundle: T[PathRef] = Task {
+    val bundleToolJar = Task.dest / "bundleTool.jar"
+    val bundleFile = Task.dest / "bundle.aab"
+    val zipPath = androidBundleZip().path
+
+    // Download bundle tool and generate .aab using the bundle zip
+    os.write(bundleToolJar, requests.get(bundleToolUrl()).bytes)
+    os.call((
+      "java",
+      "-jar",
+      bundleToolJar,
+      "build-bundle",
+      s"--modules=${zipPath}",
+      s"--output=${bundleFile}"
+    ))
+
+    PathRef(bundleFile)
+  }
+
+  /**
+   * Signs the Android App Bundle (AAB) using the provided keystore.
+   *
+   * @return PathRef to the signed app bundle.
+   */
+  def signBundle: T[PathRef] = Task {
+    val signedBundle = Task.dest / "signed_bundle.aab"
+
+    // Use jarsigner to sign the AAB file
+    os.call((
+      "jarsigner",
+      "-verbose",
+      "-sigalg",
+      "SHA256withRSA",
+      "-digestalg",
+      "SHA-256",
+      "-storepass",
+      "android",
+      "-keystore",
+      androidKeystore().path,
+      "-signedjar",
+      signedBundle,
+      androidBundle().path,
+      "androidkey"
+    ))
+
+    PathRef(signedBundle)
+  }
+
+  /**
+   * Generates a universal APK from the signed Android App Bundle (AAB).
+   *
+   * @return PathRef to the generated universal APK.
+   */
+  def androidApk: T[PathRef] = Task {
```

Comment:
Why we created bundle ? 

To generate "apks" right, I know you will say that its for uploading straight to playstore or apple store but its our duty to show and check the validation of bundle from our side so that it will not cause error during uploading to stores 
And we are providing the universal apk not the device specific apk so think this should be done coz if usee created bundle then they should be able to create apk from it...

Rest what you say will be my priority...

Reply:
I mean you can upload bundle to Play Store directly as `aab` file. And you can generate APK as well without going through the bundle flow. So what is the exact use-case of generating APK from AAB?

Reply:
TBH I don't know what the use-case will be 😅

What you say?
- Remove Apk Generation using Bundle (no use-case)

My Doubt:
- Should we sign the bundle or not?